src/jdk.zipfs/share/classes/jdk/nio/zipfs/ZipFileSystem.java

```diff
@@ -1471,6 +1471,14 @@ private void createVersionedLinks(int version) {
         };
     }
 
+    /**
+     * Package-private accessor to entry alias map used by ZipPath.
+     */
+    String lookupStringForUri(byte[] path) {
+        byte[] entry = entryLookup.apply(path);
+        return getString(entry != null ? entry : path);
```

Comment:
I think it would better to change this to lookupPath and have it return the byte[]. We can add a getRealPath in ZipPath that uses it. I think that would be more generally useful to have have a helper for URIs in ZipFileSystem.

Reply:
Addressed in https://github.com/openjdk/jdk/pull/6768/commits/e7640517dcedf2f5eebb41070e14eb9756d27d18